File: str_replace/str_replace.php
<?php
// ATOMIC_FUNCTION:  str_replace($search,$replace,$subject)
// TAGS: string
// LANG:  php
// FUNCTION: str_replace($search,$replace,$subject)

$start_val = "<body text='%body%'>";
$expected_end_val = "<body text='black'>";

$end_val = str_replace("%body%", "black", "<body text='%body%'>");

// TEST
if($end_val == $expected_end_val) {
	echo "PASS";
} else {
	
	echo "FAIL";
}

?>

File: str_replace/str_replace.java
public class str_replace {
    // ATOMIC_FUNCTION:  replaceAll(old,new)
    // TAGS: string
    // LANG:  java
    // FUNCTION: replaceAll(old,new)

    public static void main(String args[]) {
        String start_val = "<body text='%body%'>";
        String expected_end_val = "<body text='black'>";
        String end_val = start_val.replaceAll("%body%", "black");

        // TEST
        if(end_val.equals(expected_end_val)) {
            System.out.println("PASS");
        } else {

            System.out.println("FAIL");
        }
    }
}
